include/diy/master.hpp

```diff
@@ -252,10 +263,19 @@ namespace diy
 
     public:
       // Communicator functionality
-      IncomingQueues&   incoming(int gid)               { return incoming_[gid].queues; }
+      IncomingQueues&   incoming(int gid)               { return incoming_[exchange_round_].map[gid].queues; }
       OutgoingQueues&   outgoing(int gid)               { return outgoing_[gid].queues; }
       CollectivesList&  collectives(int gid)            { return collectives_[gid]; }
-      size_t            incoming_count(int gid) const   { IncomingQueuesMap::const_iterator it = incoming_.find(gid); if (it == incoming_.end()) return 0; return it->second.queues.size(); }
+      size_t            incoming_count(int gid) const
+      {
+        if (incoming_.empty())
+          return 0;
+        const IncomingQueuesMap &current_incoming = incoming_.begin()->second.map;
```

Comment:
operator[] of map is not a const function. I am going to change it to incoming_.find(exchange_round_) because that is the correct way to do this.